ruff_lsp/server.py

```diff
@@ -614,7 +614,11 @@ async def _lint_document_impl(
             show_error(f"Ruff: Lint failed ({result.stderr.decode('utf-8')})")
         return []
 
-    return _parse_output(result.stdout) if result.stdout else []
+    return (
+        _parse_output(result.stdout, settings.get("showSyntaxErrors", True))
```

Comment:
We use `settings.get` so this change is backwards compatible i.e., the new `ruff-lsp` version can be used with an old VS Code extension version.